common/appointment/decline.ts

```diff
@@ -52,4 +53,12 @@ export async function declineAppointment(user: User, appointment: Appointment) {
     }
 
     logger.info(`User(${user.userID}) declined Appointment(${appointment.id})`);
+
+    if (appointment.subcourseId) {
+        const isOnlyAppointment = (await prisma.lecture.count({ where: { subcourseId: appointment.subcourseId, isCanceled: false } })) === 1;
+        if (isOnlyAppointment) {
+            await leaveSubcourse(await prisma.subcourse.findUniqueOrThrow({ where: { id: appointment.subcourseId } }), pupil);
```

Comment:
What happens if `pupil` is null? Can't `declineAppointment` also be called with a student user?

Reply:
Good point, I think for now students are always organizers and thus see the "cancel appointment" button instead of the "decline appointment". But in fact the GraphQL Mutation is also released to OWNER, so better to support this also here (it wasn't really supported before)